x/evm/genesis.go

```diff
@@ -51,12 +52,21 @@ func InitGenesis(
 
 		code := common.Hex2Bytes(account.Code)
 		codeHash := crypto.Keccak256Hash(code)
-		if !bytes.Equal(ethAcct.GetCodeHash().Bytes(), codeHash.Bytes()) {
-			panic("code don't match codeHash")
+		isEmptyCodeHash := codeHash.Hex() == common.BytesToHash(types.EmptyCodeHash).String()
```

Comment:
use `bytes.Equal` to avoid unnecessary hex conversion

Reply:
it should be trivial to update the code hash to the correct value in the genesis prepare script. I'm not sure if it's worth trading validation to convenience to manually modifying the genesis file.

Reply:
Hey guys, sorry this totally fell off my radar, if the codehash doesn't match I believe it will cause problems down the road, so a sanity check at genesis is probably not the worst idea in the world.